torchmdnet/models/tensornet.py

```diff
@@ -227,8 +234,13 @@ def forward(
         # I avoid dividing by zero by setting the weight of self edges and self loops to 1
         edge_vec = edge_vec / edge_weight.masked_fill(mask, 1).unsqueeze(1)
         X = self.tensor_embedding(zp, edge_index, edge_weight, edge_vec, edge_attr)
+        # Total charge q is a molecule-wise property. We transform it into an atom-wise property, with all atoms belonging to the same molecule being assigned the same charge q
+        if q is None:
+            q = torch.zeros_like(z, device=z.device, dtype=z.dtype)
+        else:
+            q = q[batch]
```

Comment:
q is never None here, right?

Reply:
Also you already set up q in 225

Reply:
I understand that if charge = False in the yaml, q will be None. Or it really never goes through the representation model then?
And the q in 225 is when static shapes is True.

Reply:
in line 220 you check if q is None, then both branches define q, I do not see how q can be None in line 238 then.
I also do not understand why you need q = q[batch] in line 241. Wont it fail because you did not store batchp?
As far as I can tell these lines should just go.